### ISSUE TITLE
AddDistributedTokenCaches without enabling MicrosoftIdentityWebAppAuthentication

### ISSUE THREAD

**Issue by gerb0n** [AzureAD/microsoft-identity-web#590]
How to make use of the **distributed sqlserver cache** without enabling **MicrosoftIdentityWebAppAuthentication**?

I have a ConfidentialClientApplication which I created using this:

````
public MsalAuthProvider(IMsalTokenCacheProvider tokenCacheProvider)
{
    _app = ConfidentialClientApplicationBuilder.Create("<clientid>")
            .WithClientSecret("<clientsecret>")
            .WithAuthority("<authority>")
            .WithRedirectUri("<redirecturi>")
            .Build();

    tokenCacheProvider.InitializeAsync(_app.AppTokenCache);
    tokenCacheProvider.InitializeAsync(_app.UserTokenCache);
}
````
In the startup you can register the `IMsalTokenCacheProvider` using:
````
services.AddMicrosoftIdentityWebAppAuthentication()
    .EnableTokenAcquisitionToCallDownstreamApi()
    .AddDistributedTokenCaches();

services.AddDistributedSqlServerCache(options => { //options... });
````

I want to use the sql server cache but I do not want to enable authentication towards azuread because it conflicts with my own authentication towards my own IdentityServer

````
services.AddAuthentication("Bearer")
    .AddJwtBearer("Bearer", options => { //options...});
````

Can I register / inject the `IMsalTokenCacheProvider` some other way so I can prevent the usage of `AddMicrosoftIdentityWebAppAuthentication`?

**Comment by jmprieur** [comment]
@jennyf19 : where in fact it was useful to have the ServiceCollection extension for scenarios where customers/partners want to leverage MSAL.NET directly.

**Comment by jmprieur** [comment]
@gerb0n : you could always use `AddMicrosoftIdentityWebApiAuthentication` (as you use JwtBearer, I'm assuming this is a Web API, not a Web app), but then override what you need: getting inspired by https://github.com/AzureAD/microsoft-identity-web/wiki/customization.

Out of curiosity, what does Microsoft.Identity.Web provide that does not work for you?

**Comment by gerb0n** [comment]
@jmprieur It indeed is a Web API. It all worked fine untill I added `app.UseAuthentication()` to the `startup.cs`. 
On startup I get an internal server error `OptionsValidationException: IDW10106: The 'ClientId' option must be provided.`

I understand why I get this message because the configuration I pass into `services.AddMicrosoftIdentityWebAppAuthentication()` is empty; but then again I don't want to use **that** authentication. I want to use authentication with my JwtBearer.

Thanks for the customization link. Not sure how to apply it but I am going to find out, thanks!

**Comment by jmprieur** [comment]
@jennyf19, @henrik-me, as discussed let's take it.
I also benefits to MSAL.NET users wanting to use net472

**Comment by jennyf19** [comment]
cool. we can release it after the MSAL .NET release then.

**Comment by jennyf19** [comment]
Included in [1.4 Release](https://github.com/AzureAD/microsoft-identity-web/releases/tag/1.4.0).
